Nitrox.Bootloader/Main.cs

```diff
@@ -0,0 +1,109 @@
+
+namespace Nitrox.Bootloader
+{
+    public static class Main
+    {
+        private static readonly Lazy<string> nitroxLauncherDir = new Lazy<string>(() =>
+        {
+            string nitroxAppData = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Nitrox");
+            if (!Directory.Exists(nitroxAppData))
+            {
+                return null;
+            }
+            string nitroxLauncherPathFile = Path.Combine(nitroxAppData, "launcherpath.txt");
+            if (!File.Exists(nitroxLauncherPathFile))
+            {
+                return null;
+            }
+
+            try
+            {
+                string valueInFile = File.ReadAllText(nitroxLauncherPathFile).Trim();
+                Task.Factory.StartNew(() =>
+                {
+                    try
+                    {
+                        // Delete the path so that the launcher should be used to launch Nitrox
+                        File.Delete(nitroxLauncherPathFile);
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine($"Unable to delete the launcherpath.txt file. Nitrox will launch again without launcher. Error:{Environment.NewLine}{ex}");
+                    }
+                });
+                return Directory.Exists(valueInFile) ? valueInFile : null;
+            }
+            catch
+            {
+                // ignored
+            }
+            return null;
+        });
+        
+        public static void Execute()
+        {
+            string error = ValidateNitroxSetup();
+            if (error != null)
+            {
+                Console.WriteLine(error);
+                return;
+            }
+
+            Environment.SetEnvironmentVariable("NITROX_LAUNCHER_PATH", nitroxLauncherDir.Value);
+
+            AppDomain.CurrentDomain.AssemblyResolve += CurrentDomainOnAssemblyResolve;
+            AppDomain.CurrentDomain.ReflectionOnlyAssemblyResolve += CurrentDomainOnAssemblyResolve;
+
+            BootstrapNitrox();
+        }
+        
+        private static void BootstrapNitrox()
+        {
+            Assembly core = Assembly.Load(new AssemblyName("NitroxPatcher"));
+            Type mainType = core.GetType("NitroxPatcher.Main");
+            mainType.InvokeMember("Execute", BindingFlags.Public | BindingFlags.Static | BindingFlags.InvokeMethod, null, null, null);
+        }
+
+        
+        private static string ValidateNitroxSetup()
+        {
+            if (nitroxLauncherDir.Value == null)
+            {
+                return "Nitrox launcher path not set in AppData. Nitrox will not start.";
+            }
+            if (AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.FullName.StartsWith("QMod", StringComparison.Ordinal)) != null)
+            {
+                return "Nitrox will not start because QModManager is active.";
+            }
+
+            return null;
+        }
```

Comment:
Imo this whole return message thing and checking for it is more complicated as it needs to be.
```suggestion
        private static bool ValidateNitroxSetup()
        {
            if (nitroxLauncherDir.Value == null)
            {
                Console.WriteLine("Nitrox launcher path not set in AppData. Nitrox will not start.");
                return true;
            }
            if (AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.FullName.StartsWith("QMod", StringComparison.Ordinal)) != null)
            {
                Console.WriteLine("Nitrox will not start because QModManager is active.");
                return true;
            }

            return false;
        }
```
That way you won't need to check for null

Reply:
Your way has more code in it and couples the console type to the method. Returning the error message is simple enough imo.